src/common/store-update-message-distributor.ts

```diff
@@ -43,16 +45,14 @@ export class StoreUpdateMessageDistributor {
 
     private isValidMessage(message: StoreUpdateMessage<any>): boolean {
         return (
-            message.isStoreUpdateMessage &&
+            message.messageType === GenericStoreMessageTypes.storeStateChanged &&
             message.storeId &&
             message.payload &&
             (this.isMessageForCurrentTab(message) || message.storeType === StoreType.GlobalStore)
         );
     }
 
     private isMessageForCurrentTab(message: StoreUpdateMessage<any>): boolean {
-        return (
-            this.tabId == null || message.tabId === this.tabId // tabid will be null on inital state in content script of target page
-        );
+        return _.isEmpty(this.tabId) || message.tabId === this.tabId;
```

Comment:
I don't think `isEmpty` is right here; it returns `true` for any integer. I think the old check was correct, just the old comment was off.

If this passes tests I think it indicates that there's a missing test case, though.